virtual_rainforest/models/abiotic/atmospheric_co2.py

```diff
@@ -0,0 +1,235 @@
+        )
+
+    else:
+        plant_net_co2_assimilation1 = data["plant_net_co2_assimilation"]
+        plant_net_co2_assimilation = plant_net_co2_assimilation1.rename(
+            {"canopy_layers": "atmosphere_layers"}
+        )
+
+    # check if soil_respiration in data
+    if "soil_respiration" not in data:
+        soil_respiration = DataArray(
+            np.zeros((len(data.grid.cell_id))),
+            dims=["cell_id"],
+        )
+    else:
+        soil_respiration = data["soil_respiration"]
+
+    # check if animal_respiration in data
+    if "animal_respiration" not in data:
+        animal_respiration = DataArray(
+            np.zeros((len(data.grid.cell_id))),
+            dims=["cell_id"],
+        )
+    else:
+        animal_respiration = data["animal_respiration"]
+
+    # initialise CO2 profile
+    initial_co2_profile = initialise_co2_profile(
+        ambient_atmospheric_co2=ambient_atmospheric_co2,
+        atmosphere_layers=atmosphere_layers,
+        initialisation_method=initialisation_method,
+    )
+
+    # Calculate CO2 within canopy
+    co2_within_canopy = calculate_co2_within_canopy(
+        initial_co2_profile=initial_co2_profile.isel(
+            atmosphere_layers=slice(1, atmosphere_layers - 1)
+        ),
+        plant_net_co2_assimilation=plant_net_co2_assimilation,
+    )
+    # Calculate CO2 below canopy
+    co2_below_canopy = calculate_co2_below_canopy(
+        initial_co2_profile=initial_co2_profile.isel(atmosphere_layers=-1),
+        soil_respiration=soil_respiration,
+        animal_respiration=animal_respiration,
+    )
+    # Mix CO2
+    co2_profile_mixed = vertical_mixing_co2(
+        co2_above_canopy=initial_co2_profile.isel(atmosphere_layers=0),
+        co2_within_canopy=co2_within_canopy,
+        co2_below_canopy=co2_below_canopy,
+        mixing=False,
+    )
+
+    return co2_profile_mixed
+
+
+# helper functions
+def initialise_co2_profile(
+    ambient_atmospheric_co2: DataArray,
+    atmosphere_layers: int,
+    initialisation_method: str = "homogenous",
+) -> DataArray:
+    r"""Initialise CO\ :sub:`2`\ profile.
+
+    Args:
+        ambient_atmospheric_co2: ambient CO\ :sub:`2`\ concentraion, [ppm],
+        atmosphere_layers: number of atmosphere layers for which CO\ :sub:`2`\
+             concentration is calculated
+        initialisation_method: interpolation method for initial CO\ :sub:`2`\ profile,
+            default copies ambient CO\ :sub:`2`\ to all vertical levels
+
+    Returns:
+        initial vertical CO\ :sub:`2`\ profile, [ppm]
+    """
+
+    if initialisation_method != "homogenous":
+        raise (NotImplementedError("This method is not implemented"))
+
+    else:
+        initial_co2_profile = DataArray(
+            ambient_atmospheric_co2.expand_dims(
+                dim={"atmosphere_layers": atmosphere_layers}
+            )
+        )
+    return initial_co2_profile
+
+
+def calculate_co2_within_canopy(
```

Comment:
Worth testing here imo as the outcome is dependant on the order of subtraction so could potentially cause a sign error down the line (if the code is changed)